omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/summary/RegistrationSummaryFragmentController.java

```diff
@@ -48,12 +54,22 @@ public void controller(FragmentConfiguration config,
             app = appFrameworkService.getApp((String) config.get("appId"));
         }
         model.addAttribute("appId", app !=null ? app.getId() : "referenceapplication.registrationapp.registerPatient");
-
-        List<Extension> registrationFragments = appFrameworkService.getExtensionsForCurrentUser("registrationSummary.contentFragments", appContextModel);
-        Collections.sort(registrationFragments);
-        model.addAttribute("registrationFragments", registrationFragments);
-
+       
+        List<Extension> firstColumnFragments = appFrameworkService.getExtensionsForCurrentUser("registrationSummary.contentFragments", appContextModel);
         List<Extension> secondColumnFragments = appFrameworkService.getExtensionsForCurrentUser("registrationSummary.secondColumnContentFragments", appContextModel);
+        
+        if (CollectionUtils.isEmpty(firstColumnFragments) || CollectionUtils.isEmpty(secondColumnFragments)) {
```

Comment:
Shouldn't this be an && ?

Also, can we add a one-line comment here, explaining what this does? ("If specific registration summary widgets have not been defined, we generate our own based on the app config", or something like that.)

Reply:
> What about the use case where someone specifies their own fragments AND wants the automatically generated one?

Unfortunately it's currently the annoying all or non generation use-case here. But this would help avoid potential duplication of widgets.

> Shouldn't this be an && ?

For the case where only summary widgets for either `columnFragments` have been filled, it's then dictated by the user that they wants their summary widgets displayed and that it doesn't require for both `columnFragments` populated.

> Also, can we add a one-line comment here, explaining what this does?

Sure @mogoodrich , I am going to add this comment.

Reply:
@Ruhanga you didn't address the question of @mogoodrich about the `&&` vs `||`. Also this should have been covered by some unit tests...

Reply:
A user can decide to specify only either column `summaryWidgets` which is why I think this should be `||` rather than `&&`.

Reply:
I though the logic was we only auto-generate the fragments when the user hasn't specified *any* fragments, not when the user leaves only one column empty? @mks-d ?

Reply:
Ohh my bad, thank you very much for clearing this out @mogoodrich , yes this should be a `&&` to be explicit on all or none.